### DISCUSSION TITLE
Running OpenTelemetry without the collector

### DISCUSSION THREAD

**Discussion by iles-2e** [open-telemetry/community#1364] [Community Discussions]
We're trying to introduce tracing into several of our services written in Java. In the first phase, we would like to utilize traces to link log statements that are distributed across our services. We rely on Camel, JMS & Log4j2 among others, and to my understanding, all have good support for OpenTelemetry. But we still do not have any collector chosen to actually collect the traces themselves.

The problem is that our application refuses to start without javaagent being able to connect to a collector. We do not wish to rely on any external system, especially if it is not mission critical. We write our applications so they can run (start) and operate without a JMS broker or even without a database and (re)connect when a resource becomes available. In a gateway deployment, would this prevent any system from even starting up?

What am I missing here? I know it's probably a noobish question but I cannot find a way around this. Ideally tracing should work whether someone's actually listening or not. Like exposing metrics.. if nobody scrapes the metrics (say Prometheus) it is not interfering with the application itself.

Thank you :)

**Comment by trask** [comment]
> The problem is that our application refuses to start without javaagent being able to connect to a collector.

hi @iles-2e! can you open an issue for this at https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues?

this is very unexpected (and serious) if the javaagent is preventing an application from starting (regardless of any problems the javaagent may have internally)

thanks!

**Reply by iles-2e** [reply]
Thank you for verifying it must run without any dependency on the external system. After ruling that out, I quickly found a typo in my classpath definition, so all good!
